src/components/Admin/Admin.js

```diff
@@ -34,109 +35,145 @@ class Admin extends React.Component {
-                  <TabPanel>
-                    <h2>Streams</h2>
-                    <div className="adminTable">
-                      {
-                        this.props.settings && this.props.settings.properties && this.state.index === STREAM_TAB ?
-                        <StreamEditor {...this.props}/> : undefined
-                      }
-                    </div>
-                  </TabPanel>
-                </Tabs>
+                    </TabPanel>
+                    <TabPanel>
+                      <h2>Event Import</h2>
+                      <div className="adminTable">
+                        {
+                          this.props.settings && this.props.settings.properties && this.state.index === CUSTOM_EVENTS_TAB ?
+                          <CustomEventsEditor {...this.props}/> : undefined
+                        }
+                      </div>
+                    </TabPanel>
+                    <TabPanel>
+                      <h2>Trusted Sources</h2>
+                      <div className="adminTable">
+                        {
+                          this.props.settings && this.props.settings.properties && this.state.index === TRUSTED_SOURCES ?
+                          <div>
+                            <TrustedSources {...this.props}/>
+                          </div> : undefined
+                        }
+                      </div>
+                    </TabPanel>
+                    <TabPanel>
+                      <h2>Blacklisted Terms</h2>
+                      <div className="adminTable">
+                        {
+                          this.props.settings && this.props.settings.properties && this.state.index === BLACKLIST_TAB ?
+                          <BlacklistEditor {...this.props}/> : undefined
+                        }
+                      </div>
+                    </TabPanel>
+                    <TabPanel>
+                      <h2>Streams</h2>
+                      <div className="adminTable">
+                        {
+                          this.props.settings && this.props.settings.properties && this.state.index === STREAM_TAB ?
+                          <StreamEditor {...this.props}/> : undefined
+                        }
+                      </div>
+                    </TabPanel>
+                  </Tabs>
+                  <div className="row adminContainer">
+                    <Button className="pull-right" bsStyle="danger" onClick={this.openModal}>Restart Pipeline</Button>
+                  </div>
+                </div>
               </div>
             </div>
           </div>
         </div>
+        <Modal show={this.state.showModal} onHide={this.closeModal} aria-labelledby="contained-modal-title-sm">
+          <Modal.Header closeButton>
+            <Modal.Title id="contained-modal-title-sm">Restart Pipeline</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            Are you sure you want to restart the pipeline?
```

Comment:
Maybe put an info text that tells the user what consequences restarting the pipeline entails?

Reply:
Good idea!